test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,679 @@
+                else:
+                    new_height = max_size
+                    new_width = int(new_height * ratio)
+
+        else:
+            new_height, new_width = size
+
+        assert F.get_spatial_size(output) == [new_height, new_width]
+
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    @pytest.mark.parametrize("interpolation", INTERPOLATION_MODES)
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_image_tensor(self, size, interpolation, use_max_size, antialias, dtype, device):
+        if not (max_size_kwarg := self._make_max_size_kwarg(use_max_size=use_max_size, size=size)):
+            return
+
+        check_kernel(
+            F.resize_image_tensor,
+            self._make_input(datapoints.Image, dtype=dtype, device=device),
+            size=size,
+            interpolation=interpolation,
+            **max_size_kwarg,
+            antialias=antialias,
+            # The `InterpolationMode.BICUBIC` implementation on CUDA does not match PILs implementation well. Thus,
+            # instead of testing with an enormous tolerance, we disable the check all together.
+            check_cuda_vs_cpu=False
+            if interpolation is transforms.InterpolationMode.BICUBIC
+            else dict(rtol=0, atol=1 / 255 if dtype.is_floating_point else 1),
+            check_scripted_vs_eager=not isinstance(size, int),
+        )
+
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    @pytest.mark.parametrize("format", list(datapoints.BoundingBoxFormat))
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.int64])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_bounding_box(self, size, format, use_max_size, dtype, device):
+        if not (max_size_kwarg := self._make_max_size_kwarg(use_max_size=use_max_size, size=size)):
+            return
+
+        bounding_box = self._make_input(datapoints.BoundingBox, dtype=dtype, device=device)
+        check_kernel(
+            F.resize_bounding_box,
+            bounding_box,
+            spatial_size=bounding_box.spatial_size,
+            size=size,
+            **max_size_kwarg,
+            check_scripted_vs_eager=not isinstance(size, int),
+        )
+
+    @pytest.mark.parametrize(
+        "dtype_and_make_mask", [(torch.uint8, make_segmentation_mask), (torch.bool, make_detection_mask)]
+    )
+    def test_kernel_mask(self, dtype_and_make_mask):
+        dtype, make_mask = dtype_and_make_mask
+        check_kernel(F.resize_mask, make_mask(dtype=dtype), size=self.OUTPUT_SIZES[-1])
+
+    def test_kernel_video(self):
+        check_kernel(F.resize_video, self._make_input(datapoints.Video), size=self.OUTPUT_SIZES[-1], antialias=True)
```

Comment:
Not saying we should, but why are we parametrizing a ton for the image and bbox kernel checks, but not for masks and videos?

Reply:
Since we are actively trying to keep the number of tests low, I've opted to test very little here, because the kernels are basically just thin wrappers around the well tested image kernel:

https://github.com/pytorch/vision/blob/fc83b280387d92fe6aea73f35c1ed4ad1bb739da/torchvision/transforms/v2/functional/_geometry.py#L260-L272

https://github.com/pytorch/vision/blob/fc83b280387d92fe6aea73f35c1ed4ad1bb739da/torchvision/transforms/v2/functional/_geometry.py#L293-L300